enhancements/windows-containers/container-runtime-containerd.md

```diff
@@ -0,0 +1,213 @@
+from the previous version, up to the latest version available in the new
+cluster. For further details please refer [WMCO-Upgrade](https://github.com/openshift/enhancements/blob/master/enhancements/windows-containers/windows-machine-config-operator-upgrades.md)
+
+The procedure for an upgrade is as follows:
+1) MachineSet
+   * As part of upgrade Machine object gets deleted, resulting in the drain and deletion of the Windows node.
+   * New VM will be created.
+   * The upgraded WMCO instance will then be able to configure VM that will be created with containerd as 
+   default runtime.
+   * Once the Windows node joins kubernetes cluster, user can deploy pods on the Windows node.
+2) BYOH ( Bring Your Own Host)
+   * upgrade process will uninstall all components that were installed by WMCO. 
+   * As part of the de-configuring process Windows OS-specific configurations like HNS-Network that were created by WMCO 
+     would be deleted. 
+   * Once cleanup is done, reinstall workflow will install containerd along with kubelet, kube-proxy, CNI and 
+     hybrid-overlay.
+   * Kuebelet will start using containerd as default runtime.
+   * Once the Windows node joins kubernetes cluster, user can deploy pods on the Windows node.
+
+### Risks and Mitigations
+
+* If the cluster is upgraded, and the new version introduces an issue due to containerd
+  that should be addressed because docker runtime support won't be available in kubelet.
+  As we don't support downgrade, reverting to an older version is not possible. To overcome
+  any issues, we introduce this feature as part of the 5.0 community operator so that this feature
+  can be well tested before we make it default in WMCO 6.0 release.
+* If we run into any issue in containerd, we work closely with containerd open source community and try
+  to resolve this as soon as possible.
+* We are creating a new test job to test all e2e test cases in the community operator branch to capture any regression.
+  This will help us to identify any issues well before containerd becomes default runtime.
+* containerd doesn't support image-pull-progress-deadline as of now. There is a PR
+  https://github.com/containerd/containerd/pull/6150 work in progress. Until this
+  gets merged, if Windows image pull takes more time than the default value, we might
+  run into to image pull timeout error. To overcome this issue, user should login to Windows VM and then pull the
+  image first with ctr commandline tool. Upon successful image download, user can create pods.
+* Currently Windows_exporter has been used to collect metrics from Windows node. we do
+  have containerd support in https://github.com/prometheus-community/Windows_exporter/releases/tag/v0.16.0
+  As of now there is no parity mismatch between Docker CRI and containerd.
+
+### Test Plan
+
+* We will create a new test job (OKD) in the 4.10 community branch. We will pass WMCO feature flag in the test job
```

Comment:
Don't we need a test job that uses containerd against OCP 4.10 also?

Reply:
As we support only in the community edition, I thought creating one job with OKD is sufficient. If not we can create a test job with  OCP instead of OKD. Do we really need two jobs?  Just trying to think of resource usage. I am fine with adding two jobs if you are ok with overhead.

Reply:
We want to build confidence in using containerd on Windows nodes in OCP also during the 4.10 dev cycle. We can think about repurposing one of the existing jobs (maybe the ccm jobs) to run with containerd.

Reply:
@saifshaikh48 also suggested having two jobs are good( one for OKD and one for OCP).  I will go ahead and update it

Reply:
To be clear, we can update current OKD job to use containerd and update one of the existing ccm OCP jobs to also use containerd. That way we are not adding any new jobs per se.